src/browser/modules/App/App.tsx

```diff
@@ -83,180 +87,280 @@ import {
   buildConnectionCreds,
   getDesktopTheme
 } from 'browser-components/desktop-api/desktop-api.handlers'
-import { METRICS_EVENT, udcInit } from 'shared/modules/udc/udcDuck'
+import {
+  allowUdcInAura,
+  getConsentBannerShownCount,
+  METRICS_EVENT,
+  udcInit,
+  updateUdcData
+} from 'shared/modules/udc/udcDuck'
 import { useKeyboardShortcuts } from './keyboardShortcuts'
 import PerformanceOverlay from './PerformanceOverlay'
 import { isRunningE2ETest } from 'services/utils'
 import { version } from 'project-root/package.json'
+import { GlobalState } from 'shared/globalState'
+import { Bus } from 'suber'
+import {
+  usedTelemetrySettingSource,
+  TelemetrySettingSource
+} from 'shared/utils/selectors'
 export const MAIN_WRAPPER_DOM_ID = 'MAIN_WRAPPER_DOM_ID'
 
 declare let SEGMENT_KEY: string
 
-const getArgsSettingForKey = (args: any[], key: string) =>
-  args[1]?.global?.settings[key]
+type AppState = {
+  desktopAllowsCrashReporting: boolean
+  desktopAllowsUserStats: boolean
+  desktopTrackingId?: string
+}
 
-export function App(props: any) {
-  const [derivedTheme, setEnvironmentTheme] = useDerivedTheme(
-    props.theme,
-    LIGHT_THEME
-  )
-  // @ts-expect-error ts-migrate(7053) FIXME: No index signature with a parameter of type 'strin... Remove this comment to see the full error message
-  const themeData = themes[derivedTheme] || themes[LIGHT_THEME]
+export class App extends React.Component<any, AppState> {
```

Comment:
What's the reason for moving to a class component here?

Reply:
The metrics data event handler is now dependent on the redux state to know if we can send event or not, since it was registered in a useEffect, the handler would need to be unregistered and reregistered on every mount state had changed. At first I wasn't confident that would work well with segment and moved to a class component where the callback reference could be reused, but looking at the final implementation we could have stayed as a function component.